modules/siddhi-extensions/math/src/main/java/org/wso2/siddhi/extension/math/PercentileFunctionExtension.java

```diff
@@ -143,63 +144,131 @@ public void restoreState(Object[] state) {
         percentileOutputFunctionExtension.restoreState(state);
     }
 
+    /**
+     * Percentile calculation method
+     *
+     * To calculate the pth percentile (where p is any number greater than 0 or less than or equal to 100), do the
+     * following steps:
+     * 1. Order all the values in the data set from smallest to largest.
+     * 2. Multiply p percent by the total number of values, n. This number is called the index.
+     * 3. If the index obtained in Step 2 is not a whole number, round it up to the nearest whole number and go to Step
+     * 4a. If the index obtained in Step 2 is a whole number, go to Step 4b.
+     * 4a. Count the values in your data set from left to right (from the smallest to the largest value) until you reach
+     * the number indicated by Step 3. The corresponding value in your data set is the pth percentile.
+     * 4b. Count the values in your data set from left to right until you reach the number indicated by Step 2.
+     * The pth percentile is the average of that corresponding value in your data set and the value that directly
+     * follows it.
+     *
+     * @param valuesList values list
+     * @param percentile percentile (p)
+     * @return pth percentile value
+     */
+    public double getPercentileValue(List<Double> valuesList, double percentile) {
+
+        double percentileIndexTemp;
+        int percentileIndex;
+
+        // calculating percentile index
+        percentileIndexTemp = percentile * valuesList.size() / 100;
+
+        if (percentileIndexTemp % 1 == 0) {
+            percentileIndex = (int) percentileIndexTemp;
+            if (percentileIndex == valuesList.size()) {
+                return valuesList.get(percentileIndex - 1);
```

Comment:
Yes. percentileIndex can be 0 if the valuesList.size() = 1 and percentile is less than 50.0. Thanks for pointing it out. Fixed with https://github.com/wso2/siddhi/pull/189/commits/485e9b3471b78a000a97f20d010b9bebcdf361ca